src/js/components/DataTableColumns/__tests__/DataTableColumns-test.tsx

```diff
@@ -123,4 +123,84 @@ describe('DataTableColumns', () => {
       expect.objectContaining({ columns: ['size'] }),
     );
   });
+
+  test('pinned', async () => {
```

Comment:
```suggestion
  test('pinned', () => {
```
Do we need async? It doesn't seem like anything is using `await` syntax. I tried locally without async and all passed.